popup.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+let listeners = [];
+
+const tsSapCallFinishedMsg = 'call finished';
+
+chrome.runtime.sendMessage({
+	data: {
+		request: 'send logs'
+	}
+});
+
+chrome.runtime.onMessage.addListener(
+	function (request, sender, sendResponse) {
+		if (request.msg === tsSapCallFinishedMsg) {
+			const messagesElement = 'messages';
+			document.getElementById(messagesElement).innerHTML = formatServerResponse(request.data.serverResponse) +
+				document.getElementById(messagesElement).innerHTML;
+			const date = request.data.serverResponse.date;
```

Comment:
I generally tend to extract as many common parts as I can as a local variable. Because, if an option name (this is `data` here) changes, then we can only change it once where it defined, and not change all its occurrences by finding manually. Maybe in this context, it is not high likely possible, but I am speaking generally :)

Also, in case we need to call another option name on `request.data.serverResponse` later on, then we might not need to call the entire path once again, but only call the variable already defined for `request.data.serverResponse` or `request.data`. That was the reason behind `reusability`.

If I mixed the terms up with each other, sorry for that :) Like I said it is a very small suggestion, but again your call :)